### PR TITLE
Print the firrtl exception if we get one

### LINKED ISSUE #67: Print exceptions when firrtl compilation fails
Don't just say illegal firrtl when firrtl has a nice error message see Generate.scala:222

### PR DESCRIPTION
**Pull request by colinschmidt**
Fixes #67

### CODE REVIEW

File: tapeout/src/main/scala/transforms/Generate.scala [hunk 1]
```diff
@@ -218,8 +218,8 @@ sealed trait GenerateTopAndHarnessApp extends LazyLogging { this: App =>
       case x: FirrtlExecutionSuccess =>
         dump(x, tapeoutOptions.topFir, tapeoutOptions.topAnnoOut)
         x.circuitState.circuit.modules.collect{ case e: ExtModule => e }
-      case _ =>
-        throw new Exception("executeTop failed on illegal FIRRTL input!")
+      case e =>
```
Comment (jwright6323) [inline tapeout/src/main/scala/transforms/Generate.scala]: just a nit, but we should probably use the same case variable (`e` vs `x`) for both cases for clarity- one `e` is an `ExtModule` and the other is a result

File: tapeout/src/main/scala/transforms/Generate.scala [hunk 2]
```diff
@@ -243,7 +243,7 @@ sealed trait GenerateTopAndHarnessApp extends LazyLogging { this: App =>
     val harnessResult = firrtl.Driver.execute(optionsManager)
     harnessResult match {
       case x: FirrtlExecutionSuccess => dump(x, tapeoutOptions.harnessFir, tapeoutOptions.harnessAnnoOut)
-      case _ =>
+      case e => throw new Exception(s"executeHarness failed while executing FIRRTL!\n${e}")
```
Comment (jwright6323) [inline tapeout/src/main/scala/transforms/Generate.scala]: same comment although this is less bad

**Comment by colinschmidt** [comment]
Fixed reviewer feedback

File: tapeout/src/main/scala/transforms/Generate.scala [hunk 3]
```diff
@@ -218,8 +218,8 @@ sealed trait GenerateTopAndHarnessApp extends LazyLogging { this: App =>
       case x: FirrtlExecutionSuccess =>
         dump(x, tapeoutOptions.topFir, tapeoutOptions.topAnnoOut)
         x.circuitState.circuit.modules.collect{ case e: ExtModule => e }
-      case _ =>
-        throw new Exception("executeTop failed on illegal FIRRTL input!")
+      case x =>
+        throw new Exception(s"executeTop failed while executing FIRRTL!\n${e}")
```
Comment (abejgonzalez) [inline tapeout/src/main/scala/transforms/Generate.scala]:
```suggestion
        throw new Exception(s"executeTop failed while executing FIRRTL!\n${x}")
```

File: tapeout/src/main/scala/transforms/Generate.scala [hunk 4]
```diff
@@ -243,7 +243,7 @@ sealed trait GenerateTopAndHarnessApp extends LazyLogging { this: App =>
     val harnessResult = firrtl.Driver.execute(optionsManager)
     harnessResult match {
       case x: FirrtlExecutionSuccess => dump(x, tapeoutOptions.harnessFir, tapeoutOptions.harnessAnnoOut)
-      case _ =>
+      case x => throw new Exception(s"executeHarness failed while executing FIRRTL!\n${e}")
```
Comment (abejgonzalez) [inline tapeout/src/main/scala/transforms/Generate.scala]:
```suggestion
      case x => throw new Exception(s"executeHarness failed while executing FIRRTL!\n${x}")
```

**Pull request merged**